src/main/java/org/truffleruby/core/FinalizationService.java

```diff
@@ -120,7 +120,11 @@ private void createFinalizationThread() {
             while (true) {
                 final FinalizerReference finalizerReference = (FinalizerReference) threadManager.runUntilResult(null,
                         () -> finalizerQueue.remove());
-                finalizerReference.getFinalizerActions().forEach(action -> action.run());
+                    try {
+                        finalizerReference.getFinalizerActions().forEach(action -> action.run());
+                    } catch (Exception e) {
+                        // Do nothing, the finalizer thread must continue to process objects.
```

Comment:
It would be good to log these exceptions, maybe behind an option.
The formatting could be similar to what is used in AtExitManager.